client/src/components/CrossCheck/CrossCheckCriteriaForm.tsx

```diff
@@ -40,44 +40,44 @@ export interface CriteriaFormProps {
   countStar: CountState[];
   setCountStar: (newCount: CountState[]) => void;
   criteriaData: CrossCheckCriteriaData[];
-  totalPoints: number;
-  setTotalPoints: (newPoint: number) => void;
+  totalPoints?: number;
   penalty: CountState[];
   setPenalty: (newPenalty: CountState[]) => void;
   criteriaComment: CommentState[];
   setComment: (newComment: CommentState[]) => void;
-  maxScoreForTask: number;
 }
 
 export function CrossCheckCriteriaForm({
   countStar,
   setCountStar,
   criteriaData,
   totalPoints,
-  setTotalPoints,
   penalty,
   setPenalty,
   criteriaComment,
   setComment,
-  maxScoreForTask,
 }: CriteriaFormProps) {
+  const form = Form.useFormInstance();
   const penaltyData: CrossCheckCriteriaData[] =
     criteriaData?.filter(item => item.type.toLowerCase() === TaskType.Penalty).map(item => omit(item, ['point'])) ?? [];
 
-  function calculateTotalPoints() {
-    return useMemo(() => {
-      const sumPoints = +countStar
-        .map(item => item.point)
-        .reduce((prev, next) => prev + next, 0)
-        .toFixed(1);
-      const sumPenalty = penalty.map(item => item.point).reduce((prev, next) => prev + next, 0);
-      const finalPoints = sumPoints + sumPenalty;
-      setTotalPoints(finalPoints > 0 ? finalPoints : 0);
-      return finalPoints > 0 ? finalPoints : 0;
-    }, [countStar, penalty]);
-  }
+  useEffect(() => {
+    form.setFieldValue('score', totalPoints);
+  }, [totalPoints]);
+
+  useEffect(() => {
+    if (criteriaData.length === 0) return;
```

Comment:
Minor suggestion: this is shorter and more common
```suggestion
   if (!criteriaData.length) return;
```